### DISCUSSION TITLE
How to import dashboard using API

### DISCUSSION THREAD

**Discussion by gugupy** [apache/superset#25083] [General]
I am using Keycloak as my provider.

### What have I tried?
Get the access token from Keycloak and try import the dashboard via API.

Able to import dashboard using swagger UI but getting `msg:"The specified alg value is not allowed"` with a curl command which given by the Swagger UI.

Any help or suggestion that would be grateful.

**Sample image for getting csrf token:**
![image](https://github.com/apache/superset/assets/28136162/f7c1a671-3254-47ca-aae4-1696c1cc0cc0)
![image](https://github.com/apache/superset/assets/28136162/56df2363-7e9c-4d01-bc88-91a79958c018)

Thanks

**Comment by TheRealYoze** [comment]
I am having the same issue on my end. Also using Keycloak as my provider.

**Comment by thebaz73** [comment]
I also have same issue... have you resolved? if so how?
thank-you in advance

**Reply by felpasl** [reply]
- check the alg from your jwt in my case are RS256. I put this in superset config: `JWT_ALGORITHM = 'RS256'`
- the problem evolve to signed jwt error and to solve get the public key from keycloak realm config:
```
    JWT_PUBLIC_KEY = '''
      -----BEGIN PUBLIC KEY-----    
      [REDACTED]
      -----END PUBLIC KEY-----
    '''
```
- and i got an error about ID from keycloak guid against integer id from superset, again in config:
```
    from flask import g
    from flask_appbuilder.security.manager import AUTH_OAUTH
    from superset.security import SupersetSecurityManager

    class CustomSsoSecurityManager(SupersetSecurityManager):
        def oauth_user_info(self, provider, response=None):  # noqa: ARG002
            me = self.appbuilder.sm.oauth_remotes[provider].get("openid-connect/userinfo")
            me.raise_for_status()
            data = me.json()
            logging.debug("User info from Keycloak: %s", data)
            return {
                "username": data.get("preferred_username", ""),
                "first_name": data.get("given_name", ""),
                "last_name": data.get("family_name", ""),
                "email": data.get("email", ""),
                "role_keys": data.get("role_keys", []),
            }

        def load_user_jwt(self, _jwt_header, jwt_data):
            username = jwt_data["preferred_username"]
            user = self.find_user(username=username)
            if user.is_active:
                # Set flask g.user to JWT user, we can't do it on before request
                g.user = user
                return user
            return None

    CUSTOM_SECURITY_MANAGER = CustomSsoSecurityManager
```
